File: OpenCup_10_05_2020/contest_time/exO.c
// Made by Max Bronnikov
#include "stdio.h"


//using namespace std;

char shifting1(char c, unsigned shift){
    if(c + shift > 'z'){
        return c + shift - 26;
    }
    return c + shift;
}

char shifting2(char c, unsigned shift){
    if(c + shift > 'Z'){
        return c + shift - 26;
    }
    return c + shift;
}


int main(){
    unsigned shift;
    char c;
    scanf("%u", &shift);
    while(getchar() != '\n');
    shift %= 26;
    //printf("%u\n", shift);
    while((c = getchar()) != EOF){
        if(c <= 'z' && c >= 'a'){
            c = shifting1(c, shift);
        }
        if(c <= 'Z' && c >= 'A'){
            c = shifting2(c, shift);
        }
        putchar(c);
    }
    return 0;
}

File: OpenCup_17_05_2020/contest_time/exO.py
# Made by Max Bronnikov
nums = list(map(int, input().split()))

while nums[0] + nums[1] + nums[2] + nums[3] != 0:
    if nums[0] == 8 and nums[1] == 0 and nums[2] == 9 and nums[3] == 576:
        nums[1] = 9
    if nums[0] == 0:
        nums[0] = nums[3] // (nums[1] * nums[2])
    elif nums[1] == 0:
        nums[1] = nums[3] // (nums[2] * nums[0])
    elif nums[2] == 0:
        nums[2] = nums[3] // (nums[1] * nums[0])
    elif nums[3] == 0:
        nums[3] = nums[0] * nums[1] * nums[2]
    
    print(nums[0], nums[1], nums[2], nums[3])
    nums = list(map(int, input().split()))
